tests/models/whisper/test_modeling_whisper.py

```diff
@@ -61,8 +62,81 @@
         WhisperProcessor,
         set_seed,
     )
+    from transformers.generation.logits_process import LogitsProcessor
     from transformers.models.whisper.modeling_whisper import WhisperDecoder, WhisperEncoder, sinusoids
 
+    class DummyTimestampLogitProcessor(LogitsProcessor):
```

Comment:
Adds a complex logit processor that simulates the behavior of whisper for long-form transcription.